packages/block-library/src/button/edit.js

```diff
@@ -219,6 +263,64 @@ function ButtonEdit( props ) {
 	const nofollow = !! rel?.includes( NOFOLLOW_REL );
 	const isLinkTag = 'a' === TagName;
 
+	const {
+		createPageEntity,
+		userCanCreatePages,
+		lockUrlControls = false,
+	} = useSelect(
+		( select ) => {
+			if ( ! isSelected ) {
+				return {};
+			}
+
+			const { getSettings } = select( blockEditorStore );
+			const _settings = getSettings();
```

Comment:
```suggestion
			const _settings = select( blockEditorStore ).getSettings();
```

Suggestion (non-blocking): This could be a one liner.